npm/src/compiler/compiler.ts

```diff
@@ -15,9 +14,7 @@ type Wasm = typeof import("../../lib/node/qsc_wasm.cjs");
 // These need to be async/promise results for when communicating across a WebWorker, however
 // for running the compiler in the same thread the result will be synchronous (a resolved promise).
 export interface ICompiler {
-  checkCode(code: string): Promise<VSDiagnostic[]>;
```

Comment:
So this one is a breaking change then, right? i.e. quantum.microsoft.com will need to make code changes before they could update after this is merged.

Reply:
Yep. Definitely a breaking change. I can add the shims back in if we want to avoid disruption for now.

Which reminds me that I already merged a potentially breaking change here: 

https://github.com/microsoft/qsharp/blob/main/npm/src/compiler.ts#L48C9-L48C9

I changed type of `severity` from `number` to `string`. I have _not_ checked whether the website uses the value of this property (the playground doesn't use it so I thought it's probably fine)

Reply:
Easy enough to add it back in (with `@deprecated` tag).